core/src/main/scala/kafka/utils/CoreUtils.scala

```diff
@@ -252,16 +255,62 @@ object CoreUtils {
     listenerListToEndPoints(listeners, securityProtocolMap, true)
   }
 
+  def validateOneIsIpv4AndOtherIpv6(first: String, second: String): Boolean =
+    (inetAddressValidator.isValidInet4Address(first) && inetAddressValidator.isValidInet6Address(second)) ||
```

Comment:
Looking at the implementation of `isValidInet6Address` it seems to accept prefix sizes and zone ids. AFAIU it would not make sense to accept an address with a prefix here: We're expecting to ultimately bind to a single address. I'm not sure whether we need to accept/support zone/scope here either, for the same reason.

Reply:
> AFAIU it would not make sense to accept an address with a prefix here

> I'm not sure whether we need to accept/support zone/scope here either, for the same reason.

I haven't really looked into this, I would imagine that this is a different type of checking that would happen elsewhere? I guess I could look into this but to me this seems like a different feature/PR since this PR is specifically only about IPv4 vs IPv6.

wdyt?